pkg/virtual/initializingworkspaces/builder/build.go

```diff
@@ -168,7 +173,11 @@ type apiSetRetriever struct {
 }
 
 func (a *apiSetRetriever) GetAPIDefinitionSet(ctx context.Context, key dynamiccontext.APIDomainKey) (apis apidefinition.APIDefinitionSet, apisExist bool, err error) {
-	initializerName := string(key)
+	parts := strings.Split(string(key), "|")
+	if len(parts) != 2 {
+		return nil, false, fmt.Errorf("invalid API domain key, expected workspace|initializerName: %q", key)
+	}
+	initializerWorkspace, initializerName := parts[0], parts[1]
```

Comment:
typeWorkspace is IMO a better name. The initializer is not a thing, and not even unique by itself.